core_data_modules/traced_data/io.py

```diff
@@ -314,6 +314,97 @@ def import_coda_2_to_traced_data_iterable(cls, user, data, message_id_key, schem
                         Metadata(user, Metadata.get_call_location(), time.time())
                     )
 
+    @classmethod
+    def import_coda_2_to_traced_data_iterable_multi_coded(cls, user, data, message_id_key, scheme_keys, f=None):
+        """
+        Codes keys in an iterable of TracedData objects by using the codes from a Coda 2 messages JSON file.
+
+        Data which is has not been checked in the Coda file is coded using the provided nr_label
+        (irrespective of whether there was an automatic code there before).
+        Data which was previously coded as TRUE_MISSING, SKIPPED, or NOT_LOGICAL by any means is untouched.
+
+        TODO: Data which has been assigned a code under one scheme but none of the others needs to coded as NC not NR
+        TODO: Or, do this in Coda so as to remove ambiguity from the perspective of the RAs?
+
+        :param user: Identifier of user running this program.
+        :type user: str
+        :param data: TracedData objects to be coded using the Coda file.
+        :type data: iterable of TracedData
+        :param message_id_key: Key in TracedData objects of the message ids.
+        :type message_id_key: str
+        :param scheme_keys: Dictionary of (key in TracedData objects to assign labels to) ->
+                            (Scheme in the Coda messages file to retrieve the labels from)
+        :type scheme_keys: dict of str -> Scheme
+        :type scheme_keys: dict of str -> list of str
+        :param f: Coda data file to import codes from, or None. If None, assigns NOT_REVIEWED codes to everything.
```

Comment:
Why would we want the None behaviour?

Reply:
So that we can run the pipeline without having to send data through Coda first.